server/server.js

```diff
@@ -1,15 +1,818 @@
+    title:
+      "The Complete London 2012 Opening Ceremony | London 2012 Olympic Games",
+    url: "https://www.youtube.com/watch?v=4As0e4de-rI",
+    rating: 3211,
+  },
+  {
+    id: 283634,
+    title: "Learn Unity - Beginner's Game Development Course",
+    url: "https://www.youtube.com/watch?v=gB1F9G0JXOo",
+    rating: 211,
+  },
+  {
+    id: 562824,
+    title: "Cracking Enigma in 2021 - Computerphile",
+    url: "https://www.youtube.com/watch?v=RzWB5jL5RX0",
+    rating: 111,
+  },
+  {
+    id: 442452,
+    title: "Coding Adventure: Chess AI",
+    url: "https://www.youtube.com/watch?v=U4ogK0MIzqk",
+    rating: 671,
+  },
+  {
+    id: 536363,
+    title: "Coding Adventure: Ant and Slime Simulations",
+    url: "https://www.youtube.com/watch?v=X-iSQQgOd1A",
+    rating: 76,
+  },
+  {
+    id: 323445,
+    title: "Why the Tour de France is so brutal",
+    url: "https://www.youtube.com/watch?v=ZacOS8NBK6U",
+    rating: 73,
+  },
+];
 
-// GET "/"
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
   res.send({ express: "Your Backend Service is Running" });
 });
+
+app.get("/videos", (req, res) => {
+  const order = req.query.order || "desc";
+
+  let orderedVideos = [...videos];
+  if (order === "asc") {
+    orderedVideos.sort((a, b) => a.rating - b.rating);
+  } else {
+    orderedVideos.sort((a, b) => b.rating - a.rating);
+  }
+
+  res.json(orderedVideos);
+});
+
+app.post("/videos", (req, res) => {
+  const newVideo = req.body;
+  newVideo.id = videos.length + 1;
+  newVideo.uploadedDate = new Date().toISOString();
+  videos.push(newVideo);
+  res.json({ id: newVideo.id });
+});
+
+app.post("/videos/:id/upvote", (req, res) => {
+  const id = parseInt(req.params.id);
+  const video = videos.find((v) => v.id === id);
+  if (video) {
+    video.rating += 1;
+    res.json({});
+  } else {
+    res.status(404).json({ result: "failure", message: "Video not found" });
+  }
+});
+
+app.post("/videos/:id/downvote", (req, res) => {
+  const id = parseInt(req.params.id);
+  const video = videos.find((v) => v.id === id);
+  if (video) {
+    video.rating = Math.max(video.rating - 1, 0);
+    res.json({});
+  } else {
+    res.status(404).json({ result: "failure", message: "Video not found" });
+  }
+});
+
+app.delete("/videos/:id", (req, res) => {
+  const id = parseInt(req.params.id);
+  const initialLength = videos.length;
+  videos = videos.filter((v) => v.id !== id);
+  if (videos.length < initialLength) {
+    res.json({});
+  } else {
+    res.status(404).json({ result: "failure", message: "Video not found" });
+  }
+});
+
+// Serve static files from the 'build' folder (React frontend)
+app.use(express.static(path.join(__dirname, "build")));
+
+// Serve the index.html file as the default route
+app.get("*", (req, res) => {
+  res.sendFile(path.join(__dirname, "build", "index.html"));
+});
+
+app.listen(port, () => console.log(`Listening on port ${port}`));
+
+// update6
```

Comment:
Please avoide pushing comments which are for yourself, and not info for other devs.